tests/e2e/core-tests/specs/api/coupon.test.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable jest/no-export, jest/no-disabled-tests */
+/**
+ * Internal dependencies
+ */
+const { HTTPClientFactory, Coupon } = require( '@woocommerce/api' );
+
+/**
+ * External dependencies
+ */
+const config = require( 'config' );
+const {
+	it,
+	describe,
+	beforeAll,
+} = require( '@jest/globals' );
+
+/**
+ * Create the default coupon and tests interactions with it via the API.
+ */
+const runCouponApiTest = () => {
+	describe('REST API > Coupon', () => {
+		let client;
+		let defaultCoupon;
+		let coupon;
+		let repository;
+
+		beforeAll(async () => {
+			defaultCoupon = config.get( 'coupon' );
+			const admin = config.get( 'users.admin' );
+			const url = config.get( 'url' );
+
+			client = HTTPClientFactory.build( url )
+				.withBasicAuth( admin.username, admin.password )
+				.withIndexPermalinks()
+				.create();
+		} );
+
+		it('can create a coupon', async () => {
+			repository = Coupon.restRepository( client );
+
+			// Check properties of the coupon in the create coupon response.
+			coupon = await repository.create( defaultCoupon );
+			expect( coupon ).toEqual( expect.objectContaining( defaultCoupon ) );
+		});
+
+		it('can retrieve a coupon', async () => {
+			const couponProperties = {
+				id: coupon.id,
+				code: defaultCoupon.code,
+				discount_type: defaultCoupon.discountType,
+				amount: defaultCoupon.amount,
+			};
+
+			// Read coupon directly from API to compare.
+			const response = await client.get( `/wc/v3/coupons/${coupon.id}` );
+			expect( response.statusCode ).toBe( 200 );
+			expect( response.data ).toEqual( expect.objectContaining( couponProperties ) );
+		});
+
+		it('can update a coupon', async () => {
+			const updatedCouponProperties = {
+				amount: '75.00',
+				discount_type: 'fixed_cart',
+				free_shipping: true,
+			};
+
+			await repository.update( coupon.id, updatedCouponProperties );
+
+			// Check the coupon response for the updated values.
+			const response = await client.get( `/wc/v3/coupons/${coupon.id}` );
+			expect( response.statusCode ).toBe( 200 );
+			expect( response.data ).toEqual( expect.objectContaining( updatedCouponProperties ) );
+		});
+
+		it('can delete a coupon' ,async () => {
```

Comment:
Nice :) We'll have to go through the other api tests and add the deletes.

Reply:
Agreed! I guess for some context on the truthy check, I was hoping I could test the API results, but 400-level responses throw exceptions in the API so I am not sure if we want to adjust that in the future, or keep the behavior there as-is.

Reply:
I just noticed there is a spacing issue with that comma on line 75.

Reply:
Woops, thanks! Fixed :)